src/utils/minio_crud.py

```diff
@@ -0,0 +1,55 @@
+import os
+from minio import Minio
+from minio.error import S3Error
+
+class MinioCRUDClient:
+    def __init__(self, endpoint, access_key, secret_key, secure=True):
+        self.client = Minio(endpoint, access_key=access_key, secret_key=secret_key, secure=secure)
+
+    def create_bucket(self, bucket_name):
+        try:
+            if not self.client.bucket_exists(bucket_name):
+                self.client.make_bucket(bucket_name)
+                
+        except S3Error as exc:
+            return False
+        return True       
+
+    def list_buckets(self):
+        return self.client.list_buckets()
+
+    def write_file(self, bucket_name, object_name, file_path):
+        try:
+            self.client.fput_object(bucket_name, object_name, file_path)
+        except S3Error as exc:
+            print(f"Error writing file: {exc}")
+            return False
+        return True
```

Comment:
not a big fan of this design pattern)

Reply:
updated but let me know if there is a better pattern

Reply:
usually, if you check something and return bool after, it's just easy to return the check itself : 

like

```
if cond:
   return True
return False 
``` 

should be just 

```
return cond
```